packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.types.ts

```diff
@@ -56,8 +59,10 @@ export interface ICheckboxProps extends React.ButtonHTMLAttributes<HTMLElement |
 
   /**
    * Callback that is called when the checked value has changed.
+   * However, if the Checkbox has a controlled indeterminate state,
+   * this is called when the indeterminate value has changed.
    */
-  onChange?: (ev?: React.FormEvent<HTMLElement | HTMLInputElement>, checked?: boolean) => void;
+  onChange?: (ev?: React.FormEvent<HTMLElement | HTMLInputElement>, checked?: boolean, wasIndeterminate?: boolean) => void;
```

Comment:
Why is this needed? I do not believe there is a scenario where this matters.

Reply:
@dzearing Per our conversation I reverted onChange functionality back to only handling component checked state. I changed the example file to handle user controlled indeterminate state. Please let me know if there's anymore I can do and thanks!